library/pk_wrap.c

```diff
@@ -611,26 +611,18 @@ static void rsa_debug(mbedtls_pk_context *pk, mbedtls_pk_debug_item *items)
 }
 
 const mbedtls_pk_info_t mbedtls_rsa_info = {
-    MBEDTLS_PK_RSA,
-    "RSA",
-    rsa_get_bitlen,
-    rsa_can_do,
-    rsa_verify_wrap,
-    rsa_sign_wrap,
-#if defined(MBEDTLS_ECDSA_C) && defined(MBEDTLS_ECP_RESTARTABLE)
-    NULL,
-    NULL,
-#endif
-    rsa_decrypt_wrap,
-    rsa_encrypt_wrap,
-    rsa_check_pair_wrap,
-    rsa_alloc_wrap,
-    rsa_free_wrap,
-#if defined(MBEDTLS_ECDSA_C) && defined(MBEDTLS_ECP_RESTARTABLE)
-    NULL,
-    NULL,
-#endif
-    rsa_debug,
+    .type = MBEDTLS_PK_RSA,
```

Comment:
maybe I'm missing something here, but it seems to me that the original code only had
```
#if defined()
#endif
```
So those functions should always be `NULL` and we don't necessarily need to fill all the fields. Am I wrong?

Reply:
According to the standard you're correct, but I was worried that perhaps some compilers would complain if we don't explicitly initialize all the fields.

Reply:
Well in this case can we wait for the CI feedback before going back to fill all the fields? I like the new "shape" since it looks much clearer to me and also avoid declaring things we're not going to use.

Reply:
Clang does, in fact, complain about `a_struct_with_two_fields x = {0}`. That's why we have all these `PSA_xxx_INIT` macros instead of just telling people to write `{0}`.

Reply:
> Clang does, in fact, complain about `a_struct_with_two_fields x = {0}`. That's why we have all these `PSA_xxx_INIT` macros instead of just telling people to write `{0}`.

And I think that's also why {C23 introduced](https://www.open-std.org/jtc1/sc22/wg14/www/docs/n2900.htm) `{}`.

That said, apparently the warnings [are not always the same](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=39589) in the presence of designated initializers.

> Well in this case can we wait for the CI feedback before going back to fill all the fields? I like the new "shape" since it looks much clearer to me and also avoid declaring things we're not going to use.

I'm divided. On one hand, I was going to suggest the same myself: as long as the CI is happy we're happy. OTOH, we only have a relatively small selection of compilers on our CI (GCC, Clang, a few versions of MSVC, not even IAR), so who knows that other compilers are doing? (On the other other hand, we could wait and see if people report issues with the compiler they're using. But then again, isn't it better to try and prevent that from happening?)